sled-agent/src/artifact_store.rs

```diff
@@ -357,29 +380,29 @@ impl<T: DatasetsManager> ArtifactStore<T> {
                 }
             }
 
-            let temp_path =
-                Utf8TempPath::from_path(temp_dir.join(sha256.to_string()));
-            let file = match OpenOptions::new()
-                .write(true)
-                .create_new(true)
-                .open(&temp_path)
-                .await
+            let moved_temp_dir = temp_dir.clone();
+            let file = match tokio::task::spawn_blocking(move || {
+                camino_tempfile::Builder::new()
+                    .prefix(&format!("{sha256}."))
```

Comment:
Thoughts on using https://github.com/untitaker/rust-atomicwrites? It would also do an fsync at the end.